blocks/graph.py

```diff
@@ -161,8 +162,40 @@ def replace(self, replacements):
             substitutes.
 
         """
-        return ComputationGraph(theano.clone(self.outputs,
-                                             replace=replacements))
+        # Due to theano specifics we have to make one replacement in time
+        replacements = OrderedDict(replacements)
+        # Replacements with previous replacements. We have to track
+        # variables in the new graph corresponding to original replacements.
```

Comment:
I wanted to explain, that when I'm doing one replacement a time, I need to do `replace` in both cost computation graph and all expressions in future replacements (in keys and values of dict `replacements`).